axelrod/fingerprint.py

```diff
@@ -65,168 +28,91 @@ def __init__(self, strategy, probe=axl.TitForTat):
             A class that must be descended from axelrod.Player or an instance of
             axelrod.Player.
             Default: Tit For Tat
+        step: float = 0.01
+            The separation between each Point. Smaller steps will
+            produce more Points that will be closer together.
+        progress_bar: bool = True
+            True = show a progress bar when constructing probes and running
+            tournament.
         """
-        self.strategy = strategy
-        self.probe = probe
+        self.player = create_player(strategy)
+        self._probe_class, self._probe_kwargs = get_class_and_kwargs(probe)
+        self._step = step
+        self.progress_bar = progress_bar
+        self.points = self._create_points()
+        self._probe_list = self._create_probes()
```

Comment:
i'll have to think on it some, but here's roughly my thought process.

i thought of AshlockFingerprint as a kind of auto-generated SpacialTournament .  It has the same kinds of characteristics as a Tournament where it owns some basic data about who will play who and how and then can generate data depending on how much you want those players to play.

It owns player and probes and points (instead of edges).  it has interactions (and data).  (now that i think about it, turns, and repetitions should be in __init__ then and new_step should not be in fingerprint(except for backwards compatibility. saw that while typing))

fingerprint() method then plays the "tournament"  like tournament.play()

I found it an easier way to think about AshlockFingerprint and an easier way thoroughly test.  It split up the responsibility for creating everything needed to play and then playing/processing data.